src/main/java/org/alfresco/heartbeat/HBDataCollectorServiceImpl.java

```diff
@@ -282,16 +185,16 @@ public synchronized void onLicenseFail()
 
     private void restartAllCollectorSchedules()
     {
-        for(HBBaseDataCollector collector : collectors)
+        for( Map.Entry<String, HBBaseDataCollector>  entry : collectors.entrySet() )
```

Comment:
You can iterate here through the `collectors.values()` collection.

Reply:
Good idea, I'v made the change